src/Compression/tests/gtest_compressionCodec.cpp

```diff
@@ -1303,4 +1304,41 @@ TEST(LZ4Test, DecompressMalformedInput)
     ASSERT_THROW(codec->decompress(source, source_size, memory.data()), Exception);
 }
 
+TEST(FSSTTest, CompressDecompress)
```

Comment:
C++ unit tests are kind of underused in ClickHouse. Imho, all of what this test does can be achieved with a SQL-based test. The advantage of the latter is that it will run with all kinds of sanitizers in CI. 

Suggest to remove this test and extend the SQL test instead.